atc/exec/on_failure.go

```diff
@@ -43,3 +45,33 @@ func (o OnFailureStep) Run(ctx context.Context, state RunState) error {
 func (o OnFailureStep) Succeeded() bool {
 	return o.step.Succeeded()
 }
+
+// Set GetStep's flag "registerUponFailure".
+func (o OnFailureStep) updateGetStep(step Step) {
```

Comment:
I don't really like this; seems like it would be constant maintenance as we add more step types.